components/ndatasource/org.wso2.carbon.ndatasource.ui/src/main/java/org/wso2/carbon/ndatasource/ui/NDataSourceHelper.java

```diff
@@ -498,21 +501,29 @@ private static void setDatasourceProperties(RDBMSDSXMLConfiguration rdbmsDSXMLCo
                         }
                 }
 	}
-	
-	public static String elementToString(Element element) {
-		try {
-			if (element == null) {
-				return null;
-			}
-		    Transformer transformer = TransformerFactory.newInstance().newTransformer();
-		    StringWriter buff = new StringWriter();
-		    transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
-		    transformer.transform(new DOMSource(element), new StreamResult(buff));
-		    return buff.toString();
-		} catch (Exception e) {
-			log.error("Error while convering element to string: " + e.getMessage(), e);
-			return null;
-		}
+
+    public static String elementToString(Element element) {
+        try {
+            if (element == null) {
+                return null;
+            }
+            TransformerFactory factory = TransformerFactory.newInstance();
+            factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+            Transformer transformer = factory.newTransformer();
+            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            StringWriter buff = new StringWriter();
+            transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+            transformer.transform(new DOMSource(element), new StreamResult(buff));
+            String returnValue=buff.toString();
+            buff.close();
+            return returnValue;
+        } catch (TransformerException e) {
+            log.error("Error while convering element to string: " + e.getMessage(), e);
+            return null;
+        } catch (IOException e) {
+            log.error("Error while closing StringWriter " + e.getMessage(), e);
+            return null;
+        }
```

Comment:
@nalakas Formatting doesn't seems to be correct.

Reply:
fixed